Add more examples of how to extract block list content data (#2862)
umbraco/UmbracoDocs: Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Block-List-Editor/index.md

}
```

## Extract Block List Content data

> In some cases you might want to use the block list editor to hold some data and not necessarily render a view since the data should be presented in different areas on a page.
> An example could be a product page with variants stored in block list editor.

Reviewer: ```suggestion
In some cases, you might want to use the Block List Editor to hold some data and not necessarily render a view since the data should be presented in different areas on a page.
An example could be a product page with variants stored in a Block List Editor.
```

> In this case you can extract the variants data using the following, which returns `IEnumerable<IPublishedElement>`.

Reviewer: ```suggestion
In this case, you can extract the variant's data using the following, which returns `IEnumerable<IPublishedElement>`.
```

Example:

```csharp
@inherits Umbraco.Web.Mvc.UmbracoViewPage
@using Umbraco.Core.Models.Blocks;
@using ContentModels = Umbraco.Web.PublishedModels;
@{
    var variants = Model.Value<IEnumerable<BlockListItem>>("variants").Select(x => x.Content);
    foreach (var variant in variants)
    {
        <h4>@variant.Value("variantName")</h4>

>         <p>@variant..Value("description")</p>

Reviewer 2: I guess there should only be one dot? (.)

Author: @nielslyngsoe yep 🥇

    }
}
```

If using ModelsBuilder the example can be simplified:

Example:

```csharp
@inherits Umbraco.Web.Mvc.UmbracoViewPage
@using ContentModels = Umbraco.Web.PublishedModels;
@{
    var variants = Model.Variants.Select(x => x.Content).OfType<ProductVariant>();
    foreach (var variant in variants)
    {
        <h4>@variant.VariantName</h4>
        <p>@variant.Description</h4>
    }
}
```

> If you know the block list editor only use a single block, you can cast the collection to a specific type `T` using `.OfType<T>()` otherwise the return value willl be `IEnumerable<IPublishedElement>`.

Reviewer: ```suggestion
If you know the Block List Editor only uses a single block, you can cast the collection to a specific type `T` using `.OfType<T>()` otherwise the return value will be `IEnumerable<IPublishedElement>`.
```
